botorch/sampling/get_sampler.py

```diff
@@ -115,6 +117,14 @@ def _get_sampler_deterministic(
     return DeterministicSampler(sample_shape=sample_shape, **kwargs)
```

Comment:
If the `DeterministicPosterior` is deprecated, should we deprecate the `DeterministicSampler` as well?

Reply:
I guess it hasn't even been around long enough to be deprecated. We could just delete it if it is not needed anymore.

Reply:
So, I remove it and register again the `StochasticSampler` for the deprecated `DeterministicPosterior` or what do you recommend?

Reply:
Since `DeterministicModel` is now an `EnsembleModel`, wouldn't the `IndexSampler` be sufficient there (already dispatched for `EnsembleModel`). If that works, you can just delete `DeterministicSampler` and call it a day. If it leads to any issues, it's ok to just leave it as is.

Reply:
Yes, but the `DeterministicPosterior` is still around (but not used anymore in `DeterministicModel`), we just marked it as deprecated, but it should still have a default sampler or? The question is, which one? For this reason, I proposed to revert back to `StochasticSampler`.

Reply:
@Balandat @saitcakmak: What do we do here? Just leave it in or should I remove it? Then the question is still what will be the default sampler of the deprecated `DeterministicPosterior`? For me it is ok, to just keep it until the `DeterministicPosterior` will be removed.

Reply:
What happens if you use the `IndexSampler` with `DeterministicPosterior`? If that works without any issues, then we don't need the `DeterministicSampler` and we can just delete it (and the dispatch rule associated with it). If it leads to any issues, let's keep it for now.

Reply:
This does not work as the `IndexSampler` works with `rsample_from_base_samples` which is not implemented in the case of the `DeterministicPosterior`. So I would recommend to keep it and remove it togehter with the `DeterministicPosterior` in one of the next releases.

Reply:
Sounds good, let's do that

Reply:
Then it should be done now ;)

Reply:
Cool. I'll just add some deprecation notes to the docstrings then before merging this in.